src/lib/utils/log_manager.cpp

```diff
@@ -0,0 +1,26 @@
+#include "log_manager.hpp"
+
+#include "constant_mappings.hpp"
+#include "utils/settings/log_level_setting.hpp"
+
+namespace opossum {
+
+LogManager::LogManager(SettingsManager& settings_manager, LogLevel log_level) : _log_level(log_level) {
+  if (!settings_manager.has_setting(LOG_LEVEL_SETTING_NAME)) {
+    auto log_level_setting = std::make_shared<LogLevelSetting>(LOG_LEVEL_SETTING_NAME);
+    log_level_setting->register_at(settings_manager);
+  }
+}
+
+void LogManager::add_message(const std::string& reporter, const std::string& message, const LogLevel log_level) {
+  if (log_level < _log_level) return;
```

Comment:
What might become a problem is that creating the string could be an expensive operation in itself. Have a look at https://stackoverflow.com/questions/11832960/short-circuit-operator-output-in-c

Could you run a benchmark where you insert an ignored log statement, e.g., `log (this + chunk_id)` in  Table::get_chunk?

Reply:
I can do this. On the other hand, if we keep every message and filter while retrieving the entries, I think the overhead is something we have to accept. Not logging everything while doing performance-critical stuff is also a good idea for keeping the log small.

It is also the question what we want to log. As far as I understood it, we want to log mostly plugin actions, where we should not need to have to focus on performance too much.

Reply:
> Not logging everything while doing performance-critical stuff is also a good idea for keeping the log small.

Give people a chance to do things wrong and someone will find it ;)

Reply:
Would a comment in LogManager be enough should there even be a performance warning?

Reply:
The performance warning would simply show up the first time you log something. That's not helpful.

If we log everything, this shouldn't be as bad as you will notice the log being flooded and question the performance impact of that.